engine-src/Game/LambdaHack/Client/UI/SessionUI.hs

```diff
@@ -86,6 +86,10 @@ type ItemDictUI = EM.EnumMap ItemId LevelId
 newtype AimMode = AimMode { aimLevelId :: LevelId }
   deriving (Show, Eq, Binary)
 
+-- | Recorded in-game macro.
+newtype Macro = InGameMacro { unMacro :: [K.KM] }
```

Comment:
Actually, after this commit, this type now serves for in-game and config and content macros. So I'd probably name the type and the constructor KeyMacro or KeypressesMacro or something (I'm bad with naming).

Also, I'd write in the comment the datatype invariant. That is, which commands can't appear in the list and what order the list is in (the order in which the keys will be replayed or the reverse order. And then, we'd discover currently the invariants are not preserved. Which is good --- our types finally structure out code and force us to make code changes. :)